bindings/rust/s2n-tls/src/error.rs

```diff
@@ -22,7 +22,7 @@ pub enum ErrorType {
 }
 
 #[non_exhaustive]
-#[derive(Debug, PartialEq)]
+#[derive(Debug, Eq, PartialEq)]
```

Comment:
Since these are non_exhaustive, if any of these are public apis we could end up adding a variant that is not `Eq` and have violate backwards compatibility.